plugins/scaffolder-backend/config.d.ts

```diff
@@ -52,6 +52,15 @@ export interface Config {
      */
     EXPERIMENTAL_workspaceSerialization?: boolean;
 
+    /**
+     * Sets the provider for workspace serialization.
+     *
+     * By default, it is your database.
+     */
+    EXPERIMENTAL_workspaceSerializationProvider?:
+      | 'database'
+      | 'googleCloudBucket';
```

Comment:
I think that it should merge them together 🤞

Reply:
Maybe we can use just type "string" then?

Reply:
meh, still might want the config validation, it's pretty useful to have.

Reply:
Hm... Okie.

How it might merge them? 😅 

The idea is to have 

in [plugins/scaffolder-backend/config.d.ts]
```
EXPERIMENTAL_workspaceSerializationProvider?: 'database' 
```
and in [plugins/scaffolder-backend-module-gcp/config.d.ts]
```
EXPERIMENTAL_workspaceSerializationProvider?: 'googleCloudBucket' 
```

?

Reply:
Yep exactly, I think that if typescript does what it should, they should be merged so:

`EXPERIMENTAL_workspaceSerializationProvider` becomes: `'database' | 'googleCloudBucket'`

Reply:
Although there is a chance that i'm misunderstanding how the merging works and it doesn't work like that.

Reply:
Checked that, it seems it doesn't apply any validation.
Project starts well and I can get the value by having a value which is not listed in enum

![Screenshot 2024-06-13 at 21 40 46](https://github.com/backstage/backstage/assets/2265094/baf1c5f2-5d2a-4131-b13b-0f3a2e0091a2)

Reply:
If you're running in `dev` mode it won't validate. But you can do `yarn backstage-cli config:check --strict` or start in production mode?

Reply:
Ah okie, didn't know about that.
Seems like it doesn't merge.

So then some extra logic has to be added first to merge `config.d.ts`, cause yeah, indeed having `EXPERIMENTAL_workspaceSerializationGcpBucketName` would make sense to keep in `scaffolder-backend-module-gcp` as well

![Screenshot 2024-06-14 at 13 13 18](https://github.com/backstage/backstage/assets/2265094/15d0660d-b727-41ec-be6b-677aa891692f)